# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/serviceworkers/NavigationPreloadManager.cpp
# Commit: 40c22bd043ba
# Full Hash: 40c22bd043bac8fa0c7ee3618572b6b9f3769a76
# Author: Peter Van der Beken <peterv@propagandism.org>
# Date: 2022-05-03 15:57:51
# Description:
#   Bug 1764935 - Deal properly with Promise creation failure in NavigationPreloadManager. r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D144341
# ==============================================================================

diff -r 3f83c6e28484 -r 40c22bd043ba dom/serviceworkers/NavigationPreloadManager.cpp
--- a/dom/serviceworkers/NavigationPreloadManager.cpp	Tue May 03 08:11:27 2022 +0000
+++ b/dom/serviceworkers/NavigationPreloadManager.cpp	Tue May 03 08:23:25 2022 +0000
@@ -40,13 +40,12 @@
   return NavigationPreloadManager_Binding::Wrap(aCx, this, aGivenProto);
 }
 
-already_AddRefed<Promise> NavigationPreloadManager::SetEnabled(bool aEnabled) {
-  ErrorResult result;
-  RefPtr<Promise> promise = Promise::Create(GetParentObject(), result);
+already_AddRefed<Promise> NavigationPreloadManager::SetEnabled(
+    bool aEnabled, ErrorResult& aError) {
+  RefPtr<Promise> promise = Promise::Create(GetParentObject(), aError);
 
-  if (NS_WARN_IF(result.Failed())) {
-    result.SuppressException();
-    return promise.forget();
+  if (NS_WARN_IF(aError.Failed())) {
+    return nullptr;
   }
 
   if (!mInner) {
@@ -68,27 +67,26 @@
   return promise.forget();
 }
 
-already_AddRefed<Promise> NavigationPreloadManager::Enable() {
-  return SetEnabled(true);
+already_AddRefed<Promise> NavigationPreloadManager::Enable(
+    ErrorResult& aError) {
+  return SetEnabled(true, aError);
 }
 
-already_AddRefed<Promise> NavigationPreloadManager::Disable() {
-  return SetEnabled(false);
+already_AddRefed<Promise> NavigationPreloadManager::Disable(
+    ErrorResult& aError) {
+  return SetEnabled(false, aError);
 }
 
 already_AddRefed<Promise> NavigationPreloadManager::SetHeaderValue(
-    const nsACString& aHeader) {
-  ErrorResult result;
-  RefPtr<Promise> promise = Promise::Create(GetParentObject(), result);
+    const nsACString& aHeader, ErrorResult& aError) {
+  RefPtr<Promise> promise = Promise::Create(GetParentObject(), aError);
 
-  if (NS_WARN_IF(result.Failed())) {
-    result.SuppressException();
-    return promise.forget();
+  if (NS_WARN_IF(aError.Failed())) {
+    return nullptr;
   }
 
   if (!IsValidHeader(aHeader)) {
-    result.ThrowTypeError<MSG_INVALID_HEADER_VALUE>(aHeader);
-    promise->MaybeReject(std::move(result));
+    promise->MaybeRejectWithTypeError<MSG_INVALID_HEADER_VALUE>(aHeader);
     return promise.forget();
   }
 
@@ -111,13 +109,12 @@
   return promise.forget();
 }
 
-already_AddRefed<Promise> NavigationPreloadManager::GetState() {
-  ErrorResult result;
-  RefPtr<Promise> promise = Promise::Create(GetParentObject(), result);
+already_AddRefed<Promise> NavigationPreloadManager::GetState(
+    ErrorResult& aError) {
+  RefPtr<Promise> promise = Promise::Create(GetParentObject(), aError);
 
-  if (NS_WARN_IF(result.Failed())) {
-    result.SuppressException();
-    return promise.forget();
+  if (NS_WARN_IF(aError.Failed())) {
+    return nullptr;
   }
 
   if (!mInner) {